### PR TITLE
cellxgene download function

### PR DESCRIPTION
**Pull request by adamgayoso**
This function takes a cellxgene explorer link and downloads the corresponding h5ad

### CODE REVIEW

File: scvi/data/_built_in_data/_cellxgene.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    if file_id is None:
+        raise ValueError("Dataset not found in any collection")
+
+    url = f"{DATASETS}{dataset_id}/asset/{file_id}"
+    presigned_url = _presign_url(url)
+    _download(presigned_url, save_path, filename)
+    file_path = os.path.join(save_path, filename)
+    adata = read_h5ad(file_path)
+
+    return adata
```
Comment (watiss) [inline scvi/data/_built_in_data/_cellxgene.py:82]: Can you expose a param in the API so users who already know the `dataset_id` and/or `file_id` can just pass them in and bypass the lookup?

File: tests/dataset/test_built_in_data.py [hunk 2]
```diff
@@ -108,3 +108,9 @@ def test_download_heart_cell_atlas(save_path):
 @pytest.mark.internet
 def test_download_seurat_v4_pbmc(save_path):
     scvi.data.pbmc_seurat_v4_cite_seq(save_path=save_path, mask_protein_batches=5)
+
+
+@pytest.mark.internet
+def test_download_cellxgene(save_path):
+    url = "https://cellxgene.cziscience.com/e/4b6af54a-4a21-46e0-bc8d-673c0561a836.cxg/"
+    scvi.data.cellxgene(url, save_path=save_path)
```
Comment (watiss) [inline tests/dataset/test_built_in_data.py:116]: Validate that the path exists / maybe try to read it?

File: scvi/data/_built_in_data/_cellxgene.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+COLLECTIONS = f"{CELLXGENE_PRODUCTION_ENDPOINT}/dp/v1/collections/"
+CELLXGENE_EXPLORER = "https://cellxgene.cziscience.com/e/"
+
+
+def _presign_url(url: str):
+    resp = requests.post(url)
+    return resp.json()["presigned_url"]
+
+
+def _load_cellxgene_dataset(
```
Comment (watiss) [inline scvi/data/_built_in_data/_cellxgene.py:22]: should we link isaac's gist where this code is inspired from?

**Review by watiss — APPROVED** [review]
modulo the couple of comments I left

File: scvi/data/_built_in_data/_cellxgene.py (hunk 1 above)
Comment (adamgayoso) [reply scvi/data/_built_in_data/_cellxgene.py]: done.

File: scvi/data/_built_in_data/_cellxgene.py (hunk 3 above)
Comment (adamgayoso) [reply scvi/data/_built_in_data/_cellxgene.py:12]: I don't use anything from it anymore

File: tests/dataset/test_built_in_data.py (hunk 2 above)
Comment (adamgayoso) [reply tests/dataset/test_built_in_data.py]: done

**Pull request merged**